views/js/chatControl.js

```diff
@@ -1,9 +1,40 @@
 var chatActiveRoom = "general";
 var chatRefreshTimeMs = 30000;
 
-var refreshChat = function () {
-  document.getElementById("chat_messages").innerHTML = fetchChatHTML();
-  setChatScrollToBottom();
+const refreshChat = () => {
+  fetch('./chat/' + chatActiveRoom)
+    .then(function (response) {
+      return response.json();
+    })
+    .then(function (chatList) {
+      let messages = chatList.map((element) => {
+        return `<div class="individual_chat_message">
+          <p><span class="author">${element.name}</span>${element.message}</p>
+          </div>`;
+      });
+      document.getElementById("chat_messages").innerHTML = messages.join('\n');
+      setChatScrollToBottom();
+    });
+}
+
+const postMessageToChat = () => {
+  const name = document.getElementById("name_input").value;
+  const message = document.getElementById("message_input").value;
+  document.getElementById("message_input").value = "";
+  fetch('/chat/' + chatActiveRoom, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({
+        "name": name,
+        "message": message
```

Comment:
Does `{name, message}` work here instead?

Reply:
Yes, thanks for the catch. Done https://github.com/knative-portability/Kubercade/pull/22/commits/3c7c77f2ffdd3cf840ae14a6a216c4401e5c812f